src/App.js

```diff
@@ -5,19 +5,35 @@ import ChatLog from './components/ChatLog';
 
 const App = () => {
   const [totalLikes, setTotalLikes] = useState(0);
+  const [messages, setMessages] = useState(chatMessages); 
 
-  const handleLikeCount = (id, liked) => {
-    setTotalLikes((prevLikes) => (liked ? prevLikes + 1 : prevLikes - 1));
+  const handleLikeCount = (id) => {
+    const copyMessages = [...messages];
+    let newLiked;
+    for (let i = 0; i < copyMessages.length; i++){
+      if (copyMessages[i].id === id){
+        const oneMessage = {...copyMessages[i]}
+        oneMessage.liked = ! oneMessage.liked; 
+        copyMessages[i] = oneMessage;
+        newLiked = oneMessage.liked;
+      }
+    }
```

Comment:
If we take the approach of calculating the total liked count from the message list directly, then this logic can be simplified to focus only on updating the liked status of the clicked message. The code as written does this explicitly by duplicating the message list, locating the message that was clicked, duplicating it, updating its liked status, replacing the message in the copied list with the updated message, and setting the message.

We can write this more compactly, and more clearly (at least once we're comfortable with the map function) as follows:

```js
  const handleLikeCount = (id) => {
    setMessage(messages => (  // the current state value will be passed in to our function
      // map will make our new outer list (react needs to see a new list to redraw)
      messages.map(message => (
        // we can avoid using indexing since map passes each value one by one
        // whatever we return will end up at the "current" slot in the new array
        if (message.id === id) {
          // this is the message that was clicked, so make a copy with the liked flipped
          return {...message, liked: !message.liked};
        } else {
          // this message didn't change (it wasn't clicked), so we can return it as is
          return message;
        }
      ));
    ));
  };
```

Note the use of the callback style of `setMessages`, which is more robust to asynchronous updates since our code is given the most recent value of the piece of state, which may have diverged from the version of the data that was current at the time of the render that was captured in our callback. Since there's no async code in ChatLog, it's not _necessary_, but it's useful to get in the habit of using it.

This code may look longer, but it's primarily due to the explanatory comments.